### PR TITLE
Add favicon to docs

### PR DESCRIPTION
**Pull request by manevala**

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -108,6 +108,7 @@
 # Output file base name for HTML help builder.
 htmlhelp_basename = 'cirq_iqm-doc'
 
+html_favicon = 'images/favicon.ico'
```
Comment (smite) [inline docs/conf.py:111]:
```suggestion
html_favicon = '_static/images/favicon.ico'
```

Comment (smite) [reply docs/conf.py:111]: This is where our other projects store static images, for consistency it'd be good to do it here as well.

Comment (smite) [reply docs/conf.py:111]: `html_static_path` should be uncommented as well.

Comment (smite) [reply docs/conf.py:111]: I checked the docs, https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_favicon, and they seem to hint that the "correct" place for favicons would be `_static/`. Who knows :)

**Review by olliahonen — APPROVED** [review]

**Pull request merged**